src/server/hook_func.c

```diff
@@ -7377,10 +7385,10 @@ get_server_hook_results(char *input_file, int *accept_flag, int *reject_flag, ch
 		}
 		line_data[0] = '\0';
 	}
-	if (ndtype_flag & WRITE_NEW_NODESFILE) {
+	if (update_db == 2) {
```

Comment:
It looks like if any of the node characteristics comes out as "WRITE_NEW_NODESFILE" then WRITENODE_STATE will be ignored.
I'd suggest instead of assigning numbers to update_db variable do update_db |= ndtype_flag.
then check if (update_db & WRITE_NEW_NOESFILE) or (update_db & WRITENODE_STATE) before writing.